bench/common/blas_benchmark.hpp

```diff
@@ -75,54 +75,219 @@ const inline std::string& type_string<double>() {
   return str;
 }
 
-template <typename TimeT = std::chrono::nanoseconds,
+/**
+ * @fn time_event
+ * @brief inspects an event to calculate how long it took (start/end times).
+ * Overloaded to support sycl and opencl.
+ */
+inline cl_ulong time_event(cl::sycl::event e) {
+  // get start and ed times
+  cl_ulong start_time = e.template get_profiling_info<
+      cl::sycl::info::event_profiling::command_start>();
+
+  cl_ulong end_time = e.template get_profiling_info<
+      cl::sycl::info::event_profiling::command_end>();
+
+  // return the delta
+  return (end_time - start_time);
+}
+
+inline cl_ulong time_event(Event e) {
+  // get start and end times
+  cl_ulong start_time, end_time;
+  bool all_ok = true;
+  // declare a lambda to check the result of the calls
+  auto check_call = [&](cl_int status) {
+    switch (status) {
+      case CL_SUCCESS:
+        return;
+        break;
+      case CL_PROFILING_INFO_NOT_AVAILABLE:
+        std::cerr << "The opencl queue has not been configured with profiling "
+                     "information! "
+                  << std::endl;
+        break;
+      case CL_INVALID_VALUE:
+        std::cerr << "param_name is not valid, or size of param is < "
+                     "param_value_size in profiling call!"
+                  << std::endl;
+        break;
+      case CL_INVALID_EVENT:
+        std::cerr << "event is invalid in profiling call " << std::endl;
+        break;
+      case CL_OUT_OF_RESOURCES:
+        std::cerr << "cl_out_of_resources in profiling call" << std::endl;
+        break;
+      case CL_OUT_OF_HOST_MEMORY:
+        std::cerr << "cl_out_of_host_memory in profiling call" << std::endl;
+        break;
+    }
+    // If we've reached this point, something's gone wrong - set the error flag
+    all_ok = false;
```

Comment:
why not use the default of the switch for this?

Reply:
good point!